src/components/index.ts

```diff
@@ -1,224 +1,299 @@
+export type {AppProviderProps} from './AppProvider';
 
-export {Autocomplete, AutocompleteProps} from './Autocomplete';
+export {Autocomplete} from './Autocomplete';
+export type {AutocompleteProps} from './Autocomplete';
 
-export {Avatar, AvatarProps} from './Avatar';
+export {Avatar} from './Avatar';
+export type {AvatarProps} from './Avatar';
 
-export {Backdrop, BackdropProps} from './Backdrop';
+export {Backdrop} from './Backdrop';
+export type {BackdropProps} from './Backdrop';
 
-export {Badge, BadgeProps} from './Badge';
+export {Badge} from './Badge';
+export type {BadgeProps} from './Badge';
 
-export {Banner, BannerProps, BannerStatus} from './Banner';
+export {Banner} from './Banner';
+export type {BannerProps, BannerStatus} from './Banner';
 
-export {Breadcrumbs, BreadcrumbsProps} from './Breadcrumbs';
+export {Breadcrumbs} from './Breadcrumbs';
+export type {BreadcrumbsProps} from './Breadcrumbs';
 
-export {Button, ButtonProps, buttonFrom, buttonsFrom} from './Button';
+export {Button, buttonFrom, buttonsFrom} from './Button';
+export type {ButtonProps} from './Button';
 
-export {ButtonGroup, ButtonGroupProps} from './ButtonGroup';
+export {ButtonGroup} from './ButtonGroup';
+export type {ButtonGroupProps} from './ButtonGroup';
 
-export {CalloutCard, CalloutCardProps} from './CalloutCard';
+export {CalloutCard} from './CalloutCard';
+export type {CalloutCardProps} from './CalloutCard';
 
-export {Caption, CaptionProps} from './Caption';
+export {Caption} from './Caption';
+export type {CaptionProps} from './Caption';
 
-export {Card, CardProps} from './Card';
+export {Card} from './Card';
+export type {CardProps} from './Card';
 
-export {Checkbox, CheckboxProps} from './Checkbox';
+export {Checkbox} from './Checkbox';
+export type {CheckboxProps} from './Checkbox';
 
-export {ChoiceList, ChoiceListProps} from './ChoiceList';
+export {ChoiceList} from './ChoiceList';
+export type {ChoiceListProps} from './ChoiceList';
 
-export {Collapsible, CollapsibleProps} from './Collapsible';
+export {Collapsible} from './Collapsible';
+export type {CollapsibleProps} from './Collapsible';
 
-export {ColorPicker, ColorPickerProps} from './ColorPicker';
+export {ColorPicker} from './ColorPicker';
+export type {ColorPickerProps} from './ColorPicker';
 
-export {Connected, ConnectedProps} from './Connected';
+export {Connected} from './Connected';
+export type {ConnectedProps} from './Connected';
 
-export {ContextualSaveBar, ContextualSaveBarProps} from './ContextualSaveBar';
+export {ContextualSaveBar} from './ContextualSaveBar';
+export type {ContextualSaveBarProps} from './ContextualSaveBar';
 
-export {
-  DataTable,
+export {DataTable} from './DataTable';
+export type {
   DataTableProps,
   TableData,
   TableRow,
   SortDirection,
   ColumnContentType,
 } from './DataTable';
 
-export {DatePicker, DatePickerProps, Range, Months, Year} from './DatePicker';
+export {DatePicker, Months} from './DatePicker';
```

Comment:
I'm new to this, and find surprising that `Months` is in this export statement (rather than the one below, alongside type exports), being an `enum`. Can you explain why this is the case?

Reply:
Broadly speaking you've got two sorts of thing: "values" which have some kind of JS output, and "types" which are only used at compile time and get stripped away from the resultant JS.

[A enum ends up outputting javascript code](https://www.typescriptlang.org/play/#code/KYOwrgtgBAgiCi5oG8CwAoKWBuBDANmMAIwA0GWUehwATBgL4bPoBmYIAxgC4CWA9iCi4ATgHMAkgGdEkABSixALlgIkASijIoFLCODcwIoYsYtF02RDlwrAOmpFi6gNwsA9O6gBaXy24AngAOwLAAKsGhALxQAOSOJLFQAD5xCbSxbugY7Fx8gsLi0gCqIAIgESFyAEYqMJXAmtq6UPqGxoViZtk9FlKl5Q1y8QROsepAA) so it is a value (same as something like `const a =1`, which means it gets exported with all our other values (like constants and functions etc).